apps/aechannel/src/aesc_fsm.erl

```diff
@@ -4006,8 +4168,30 @@ block_hash_from_op(#op_min_depth{data = #op_data{block_hash = BlockHash}}) ->
     BlockHash;
 block_hash_from_op(#op_watch{data = #op_data{block_hash = BlockHash}}) ->
     BlockHash;
+block_hash_from_op(#op_close{data = #op_data{block_hash = BlockHash}}) ->
+    BlockHash;
 block_hash_from_op(?NO_OP) -> %% in case of unexpected close
     ?NOT_SET_BLOCK_HASH.
 
 tx_env_and_trees_from_top(Type) ->
     aesc_tx_env_cache:tx_env_and_trees_from_top(Type).
+
+%% This function is meant to prune and prettify stack traces
+%% containing #data{} records, which often are so large that printouts become
+%% unreadable.
+pr_stacktrace(T) ->
```

Comment:
Why not use `lager:pr_stacktrace/2` instead?

Reply:
It doesn't do quite the same thing. For one thing, it inserts line breaks, which messes with the Datadog side, and the main feature of my `pr_stacktrace/1` is that it removes some bulky record elements, which lager obviously can't do on its own.

Reply:
👍